helm-eshell.el

```diff
@@ -71,6 +77,7 @@
    (nomark :initform t)
    (persistent-action :initform 'ignore)
    (nohighlight :initform t)
+   (fuzzy-match :initform (eval helm-eshell-fuzzy-match))
```

Comment:
Instead of adding a slot here with `eval`, I would prefer adding a method here:

```elisp
(defmethod helm--setup-source ((source helm-esh-source))
  (setf (slot-value source 'fuzzy-match helm-eshell-fuzzy-match)))
```